imagetest/test_suites/metadata/shutdown_script_test.go

```diff
@@ -0,0 +1,60 @@
+
+package metadata
+
+import (
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+)
+
+const shutdownTime = 110 // about 2 minutes
+
+// TestGuestShutdownScript test that shutdown scripts can run for around two minutes
+func TestGuestShutdownScript(t *testing.T) {
+	// second boot
+	bytes, err := ioutil.ReadFile("/shutdown.txt")
+	if err != nil {
+		t.Fatalf("error reading file: %v", err)
+	}
+	lines := strings.Split(strings.TrimSpace(string(bytes)), "\n")
+	if len(lines) < shutdownTime {
+		t.Fatalf("shut down time less than %d seconds.", shutdownTime)
+	}
+}
+
+// TestShutdownScript test the standard metadata script.
+func TestShutdownScript(t *testing.T) {
+	// second boot
+	bytes, err := ioutil.ReadFile(shutdownOutputPath)
+	if err != nil {
+		t.Fatalf("failed to read shutdown output %v", err)
+	}
+	output := strings.TrimSpace(string(bytes))
+	if output != shutdownContent {
+		t.Fatalf(`shutdown script output expect "%s", but actually "%s"`, shutdownContent, output)
+	}
+}
+
+// TestRandomShutdownScriptNotCrashVM test that a script with random content
+// doesn't crash the vm.
+func TestRandomShutdownScriptNotCrashVM(t *testing.T) {
+	// second boot
+	if _, err := utils.GetMetadataAttribute("shutdown-script"); err != nil {
```

Comment:
so in this test we are testing whether setting the metadata at instance creation will result in successful get of that same metadata here in the test. i don't know how it relates to the title of this test. what would a failing test look like?

Reply:
In my understanding, the random script is more like a bad script, i.e. a bad script failed and exit doesn't crash vm in terms of make disk bad that can't reboot.

Reply:
curl metadata is just a way to make sure the instance is normal as it's first boot without running shutdown script

Reply:
how can it 'crash' the vm? so this test could do anything, it could just log 'success' and it would be a good test?

Reply:
So I rename test to TestShutdownScriptFailedNotCrashVM